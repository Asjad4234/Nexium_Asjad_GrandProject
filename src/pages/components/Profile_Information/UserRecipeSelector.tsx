import { Button } from '@headlessui/react'

interface UserPinsSelectorProps {
    displaySetting: string,
    setDisplaySetting: (arg: string) => void,
}

const unSelected = "py-1.5 px-3 hover:text-blue-600 hover:scale-105 hover:shadow text-center rounded-md h-8 text-sm flex items-center gap-1 lg:gap-2"
const selected = "text-blue-600 bg-gray-100 py-1.5 px-3 hover:text-blue-600 hover:scale-105 hover:shadow text-center rounded-md h-8 text-sm flex items-center gap-1 lg:gap-2"

export default function UserRecipeSelector({ displaySetting, setDisplaySetting }: UserPinsSelectorProps) {
    return (
        <div className="flex rounded-md w-[200px] justify-between mt-3" role="group">
            <Button
                className={displaySetting==='created' ? selected : unSelected}
                onClick={() => setDisplaySetting('created')}
            >
                Created
            </Button>
            <Button
                className={displaySetting==='favorites' ? selected : unSelected}
                onClick={() => setDisplaySetting('favorites')}
            >
                Favorites
            </Button>
        </div>
    );
}