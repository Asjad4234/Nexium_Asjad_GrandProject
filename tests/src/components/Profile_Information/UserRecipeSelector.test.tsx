import UserRecipeSelector from "../../../../src/components/Profile_Information/UserRecipeSelector";
import { render, screen, fireEvent } from '@testing-library/react'

describe('The Profile recipe view selector',()=>{
    it('renders', () => {
        const { container } = render(<UserRecipeSelector displaySetting="created" setDisplaySetting={()=>({})}/>)
        expect(container).toMatchSnapshot()
    })

    it('will switch to the created view', () => {
        const mockSetDisplaySetting = jest.fn();
        render(<UserRecipeSelector displaySetting="created" setDisplaySetting={mockSetDisplaySetting}/>)
        fireEvent.click(screen.getByText('Created'));
        expect(mockSetDisplaySetting).toHaveBeenCalledWith("created")
    })

    it('will switch to the favorites view', () => {
        const mockSetDisplaySetting = jest.fn();
        render(<UserRecipeSelector displaySetting="favorites" setDisplaySetting={mockSetDisplaySetting}/>)
        fireEvent.click(screen.getByText('Favorites'));
        expect(mockSetDisplaySetting).toHaveBeenCalledWith("favorites")
    })
})
