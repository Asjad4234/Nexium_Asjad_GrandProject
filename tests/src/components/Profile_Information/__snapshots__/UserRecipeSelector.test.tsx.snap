// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Profile recipe view selector renders 1`] = `
<div>
  <div
    class="flex rounded-md w-[200px] justify-between mt-3"
    role="group"
  >
    <button
      class="text-blue-600 bg-gray-100 py-1.5 px-3 hover:text-blue-600 hover:scale-105 hover:shadow text-center rounded-md h-8 text-sm flex items-center gap-1 lg:gap-2"
      data-headlessui-state=""
      type="button"
    >
      Created
    </button>
    <button
      class="py-1.5 px-3 hover:text-blue-600 hover:scale-105 hover:shadow text-center rounded-md h-8 text-sm flex items-center gap-1 lg:gap-2"
      data-headlessui-state=""
      type="button"
    >
      Favorites
    </button>
  </div>
</div>
`;
